### ISSUE TITLE
Errors thrown from Node vm.Script cause pool.run() to resolve to null instead of rejecting

### ISSUE THREAD

**Issue by karmeleon** [jasnell/piscina#513]
Here's a minimal repro:

```js
// index.js
const Piscina = require('piscina');
const path = require('path');

async function main() {
    const pool = new Piscina({
        filename: path.resolve(__dirname, 'worker.js'),
    });

    try {
        const result = await pool.run({ type: process.argv[2] });
        console.log(result);
    } catch (e) {
        console.log(e.toString());
    }

    await pool.destroy();
}

main();
```

```js
// worker.js
const vm = require('vm');

module.exports = ({ type }) => {
    if (type === 'vm') {
        const script = new vm.Script('throw new Error("test error from vm")', {});
        script.runInNewContext();
    } else {
        throw new Error('test error');
    }
}
```

Running `node index.js novm` causes `pool.run()` to reject with `Error('test error')` as expected. However. `node index.js vm` causes the `pool.run()` call to resolve to `null` instead. Prior to `v4.2.0`, both ways would correctly reject with the thrown error.

Tested on Ubuntu 22.04.3 using `node v20.10.0`.

**Comment by metcoder95** [comment]
Hey 👋 
I can confirm the reproduction, of the recent migration to the native `EventEmitterAsyncResource` we started checking the prototype, and as the error thrown does not share the prototype with the current context, it causes it to fallback to `null`.

Opening a PR for it now 👍

**Comment by metcoder95** [comment]
Closed by #514

**Comment by karmeleon** [comment]
Amazing, thank you so much for the instant fix!
